proposal.rst

```diff
@@ -351,8 +352,8 @@ Plan :
         
         1. Using channels one could use for-range construct.
 
-        for i := range container.Iter(){
-          i.GetData() // methods to get information from the iterator at that position.
+        for i := range doc.terms(){
+          i.GetTerm() // methods to get term from the iterator at that position.
```

Comment:
This sounds like a good way to wrap these, but what's actually involved in doing so?

Reply:
you mean to show me how its done internally in go?...If yes then ill code it and show...thats done using channels in go ... in a brief range gets data from the channel which is open until iterator is reaches the last element , then channel closes and range stops..

Reply:
> This sounds like a good way to wrap these, but what's actually involved in doing so?

Hey I have added a link how this can be done with small piece of code.

Reply:
Kind of, but that's just showing how to create a channel using only Go code.  The real (and perhaps more tricky, though perhaps not) question is what's actually involved in doing the same but with data coming from a Xapian iterator wrapped via SWIG.

Reply:
> Kind of, but that's just showing how to create a channel using only Go code. The real (and perhaps more tricky, though perhaps not) question is what's actually involved in doing the same but with data coming from a Xapian iterator wrapped via SWIG.

Hey I have put the code for implementing termiterator in document class (I added only raw code how termiterator is implemented but the other parts of document interface needs rewrapping which will be done week 2 , so please consider only for-range construct  implementation part)

Reply:
> Kind of, but that's just showing how to create a channel using only Go code. The real (and perhaps more tricky, though perhaps not) question is what's actually involved in doing the same but with data coming from a Xapian iterator wrapped via SWIG.

Hey , I have put the code for iterators , errors , type conversions , rewrapping . Please review it. Thank You 😊

Reply:
> https://github.com/srinivasyadav18/xapian-gsoc-plan/blob/master/go.i#L43

This looks promising, though it seems only provides access to the term name.

What about the other accessor methods on the `TermIterator` object?  At least for some of them, they can involve significant work so just reading all of them up front isn't great.  The Python bindings have attempted to address such issues already - maybe that's worth looking at to see if the approach works for Go too.

It's also good to try to partly automate such wrapping via SWIG macros - otherwise you're going to end with a lot of code cut-and-pasted and then modified to support all the different methods which return a `TermIterator`, and that's not great for maintainability.

Reply:
Hey that’s not a big issue ..I could just send the term iterator object over the channel ..so now through iteration we can access all the method of it.